tests/cpgrid/lookupdataCpGrid_test.cpp

```diff
@@ -229,3 +266,121 @@ BOOST_AUTO_TEST_CASE(no_lgrs_grid)
     grid.createCartesian(grid_dim, cell_sizes);
     lookup_check(grid);
 }
+
+
+void fieldProp_check(const Dune::CpGrid& grid, Opm::EclipseGrid eclGrid, std::string deck_string)
+{
+    Opm::Deck deck = Opm::Parser{}.parseString(deck_string);
+    Opm::FieldPropsManager fpm(deck, Opm::Phases{true, true, true}, eclGrid, Opm::TableManager());
+    const auto& poro = fpm.get_double("PORO");
+    // LookUpData
+    auto leaf_view = grid.leafGridView();
+    const Opm::LookUpData<Dune::CpGrid,Dune::GridView<Dune::DefaultLeafGridViewTraits<Dune::CpGrid>>> lookUpData(leaf_view);
+    // LookUpCartesianData
+    const Dune::CartesianIndexMapper<Dune::CpGrid> cartMapper(grid);
+    const Opm::LookUpCartesianData<Dune::CpGrid, Dune::GridView<Dune::DefaultLeafGridViewTraits<Dune::CpGrid>>>
+        lookUpCartesianData(leaf_view, cartMapper);
+    // Element mapper
+    const Dune::MultipleCodimMultipleGeomTypeMapper<Dune::GridView<Dune::DefaultLeafGridViewTraits<Dune::CpGrid>>>
+        mapper(leaf_view, Dune::mcmgElementLayout());
+    for (const auto& elem : elements(leaf_view))
+    {
+        const auto elemIdx = mapper.index(elem);
+        const auto elemOriginIdx = elem.getOrigin().index();
+        BOOST_CHECK_EQUAL(poro[elemOriginIdx], lookUpData.fieldPropDoubleFromEntity<Dune::cpgrid::Entity<0>>(fpm, "PORO", elem));
+        BOOST_CHECK_EQUAL(poro[elemOriginIdx], lookUpData.fieldPropDouble(fpm, "PORO", elemIdx));
+        BOOST_CHECK_EQUAL(poro[elemOriginIdx], lookUpCartesianData.fieldPropDoubleFromEntity<Dune::cpgrid::Entity<0>>(fpm, "PORO", elem));
+        BOOST_CHECK_EQUAL(poro[elemOriginIdx], lookUpCartesianData.fieldPropDouble(fpm, "PORO", elemIdx));
+    }
+}
+
+
+BOOST_AUTO_TEST_CASE(fieldProp) {
+    std::string deckString =
+        R"( RUNSPEC
+        DIMENS
+        1  1  5 /
+        GRID
+        COORD
+        0 0 0
+        0 0 1
+        1 0 0
+        1 0 1
+        0 1 0
+        0 1 1
+        1 1 0
+        1 1 1
+        /
+        ZCORN
+        4*0
+        8*1
+        8*2
+        8*3
+        8*4
+        4*5
+        /
+        ACTNUM
+        5*1
+        /
+        PORO
+        5*0.15
+        /)";
+
+    Opm::Parser parser;
+    const auto deck = parser.parseString(deckString);
+
+    Dune::CpGrid grid;
+    Opm::EclipseGrid eclGrid(deck);
+
+    grid.processEclipseFormat(&eclGrid, nullptr, false, false, false);
+
+    fieldProp_check(grid, eclGrid, deckString);
+}
+
+
+BOOST_AUTO_TEST_CASE(fieldPropLgr) {
+    std::string deckString = R"( RUNSPEC
+DIMENS
+1 1 5
+/
+GRID
+DX
+-- There are in total 1 cell with length 1ft in x-direction
+1*1
+/
+DY
+-- There are in total 1 cell with length 1ft in y-direction
+1*1
+/
+DZ
+-- The layers are 2 ft thick, in each layer there is 1 cell
+1*2 1*2 1*2 1*2 1*2
+/
+TOPS
+40*1
+/
+ACTNUM
+5*1
+/
+PORO
+5*0.15
```

Comment:
As we are checking the values in our tests, we should make sure that they differ for different cells, by using e.g.
```
PORO
1.0 2.0 3.0 4.0 5.0
/
```

If you decide to also check an in propery, then you could use EQLNUM in addition by adding this at the bottom
```
PROPS

EQLNUM
1 2 3 4 5
/
```

```

Reply:
It seems that EQLNUM is not supported, or I did something wrong before this other change.